src/components/app.tsx

```diff
@@ -298,47 +311,103 @@ export class AppComponent extends BaseComponent<IProps, IState> {
     const bottomRightLat = scenarioData[bottomRightLatKey];
     const bottomRightLng = scenarioData[bottomRightLngKey];
 
+    kTabInfo.blocks.index = showBlocks ? 0 : -1;
+    kTabInfo.code.index = showCode ? kTabInfo.blocks.index + 1 : -1;
+    kTabInfo.controls.index = showControls ? (showCode ? kTabInfo.code.index + 1 : kTabInfo.blocks.index + 1) : -1;
+    let currentTabType = SectionTypes.BLOCKS;
+    if (showCode && this.state.tabIndex === kTabInfo.code.index) {
+      currentTabType = SectionTypes.CODE;
+    }
+    if (showControls && this.state.tabIndex === kTabInfo.controls.index) {
+      currentTabType = SectionTypes.CONTROLS;
+    }
+
     return (
       <App className="app" ref={this.rootComponent}>
         <ResizeObserver
           onResize={this.resize}
         />
         <Row>
-          <Tabs>
+          <Tabs selectedIndex={this.state.tabIndex} onSelect={this.handleTabSelect}>
+            <TabBack
+              width={tabWidth}
+              backgroundcolor={this.getTabColor(currentTabType)}
+            />
             <TabList>
-              { showBlocks && <Tab>Blocks</Tab>}
-              { showCode && <Tab>Code</Tab>}
-              { showControls && <Tab>Controls</Tab>}
+              { showBlocks &&
+                <Tab
+                  borderwidth={this.state.tabIndex !== kTabInfo.blocks.index ? "2px" : "0px"}
+                  leftradius={"0px"}
+                  rightradius={this.state.tabIndex === (kTabInfo.blocks.index + 1) ? "10px" : "0px"}
+                  backgroundcolor={this.getTabColor(SectionTypes.BLOCKS)}
+                  backgroundhovercolor={this.getTabHoverColor(SectionTypes.BLOCKS)}
+                >
+                  {this.getTabName(SectionTypes.BLOCKS)}
+                </Tab>
+              }
+              { showCode &&
+                <Tab
+                  borderwidth={this.state.tabIndex !== kTabInfo.code.index ? "2px" : "0px"}
+                  leftradius={this.state.tabIndex === (kTabInfo.code.index - 1) ? "10px" : "0px"}
+                  rightradius={this.state.tabIndex === (kTabInfo.code.index + 1) ? "10px" : "0px"}
+                  backgroundcolor={this.getTabColor(SectionTypes.CODE)}
+                  backgroundhovercolor={this.getTabHoverColor(SectionTypes.CODE)}
+                >
+                  {this.getTabName(SectionTypes.CODE)}
+                </Tab>
+              }
+              { showControls &&
+                <Tab
+                  borderwidth={this.state.tabIndex !== kTabInfo.controls.index ? "2px" : "0px"}
+                  leftradius={this.state.tabIndex === (kTabInfo.controls.index - 1) ? "10px" : "0px"}
```

Comment:
Note that I went with passing the value of the CSS property in via the styled component props.  Another pass is to pass in a value that is used in some computation or comparison to obtain the property value inside of the component.  I like the flexibility of passing in an arbitrary property value without having to create a complex calculation inside of the styled component CSS.

Reply:
Generally a fan of semantic properties, so having a property like `isTerminal` or  `isFinal` `isLast` and then having the component do the right thing (based on its own metrics) would be my preference with this sort of thing, but I understand your concerns too.

Reply:
The semantic route was what I initially went with and what I would prefer for a typical React component (let the component decide what to do based on the props), but that model seems a little less intuitive with styled components and wasn't in line with some of the examples in the documentation.  This is especially true when passing in things like colors or dealing with CSS properties that have more than a binary state (the CSS code in the styled component gets ugly) like we do with some of these complex border stylings needed to get the tabs to match the spec.  However, I'm game to tweak this if others think semantic properties are the right direction.

Reply:
I wonder if you could just wrap the styled component with a simple functional component wrapper that does those calculations, and export that?  

I wouldn't sweat this though, its not going to stall a PR :)

Reply:
I tweaked the props to use semantic names and let the component determine its own styling.  I was a little thrown off by having to pass in strings (and non-camel case), but maybe someone with more experience with styled components in Typescript knows a better pattern.  Nonetheless, I think it's probably ok for now.